pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16341314?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16341314?buildTab=artifacts#/) on release-24.2 @ [456fc36c2e2447bba18f80fba6be8a723ff03982](https://github.com/cockroachdb/cockroach/commits/456fc36c2e2447bba18f80fba6be8a723ff03982):


```
    "queryArgs": [
     "schema_w1_1"
    ],
    "result": []
   }
  ],
  "previousStatements": [
   "CREATE SCHEMA IF NOT EXISTS schema_w0_13 AUTHORIZATION root",
   "INSERT INTO schema_w1_1.table_w1_11 (\"\"\"c\rol11_w1_12\",col11_w1_13,\"c ol11_w1_14\",\"c ol11_w1_15\",col11_w1_16,\"co\\gl11_w1_17\",col11_w1_18,col11_w1_19,col11_w1_20,col11_w1_21) VALUES (3802:::OID,e'\\x06:J'::NAME:::NAME,true,(-0.49315309524536133):::FLOAT8,'1976-01-12 19:52:25.000872':::TIMESTAMP,'BOX(-10 -10,10 10)':::BOX2D,(-9):::INT8,1.16785979309038:::FLOAT8,'[-1.8974963e-29,-8.8278396e+33,-0.102816574,1.776673e+16,0.0011216166,4.2506004e-26,5.2259508e+36,9.7553276e+17,-1.8732823e-20,1.5286023e-05,-2.326369e+25,8.35794e-23,2592.8896,-6.249611e-23,1.3412827e-33,-1.2094203e+25,-6.276003e+31,1.0887266e+14,-3.2851452e+26,1.2039619e+12,-1.2485945e+12,3.442424e-30,4.355172e-06,3.8542016e+17,-2.038112e+12,0.1371942,-4.291276e+34,-206.83076,4.0177918e-32,-3.7237536e-13,3074.2256,1.4580748e-06,-9.97458e-14,7.5720546e+25,-4.033352e-22,8.8600563e+33,0.006987787,2.0047403e-10,7.5545963e+12,-8.61833e+31,-0.0040332465,9.559631e+19,-5.183164e+23,-6.9620285e+19,2.106012e-12,-0.010029208,1.804266e-14,-0.00089507987,1.6780399e+29,1.2947271e-28,5.1822084e+07,20208.246,-3.187815e-27,-0.07016158,3.3800535e+13,8.769164e-30,-2.0803904e+06,4.7790184e-27,1.0531112e+27,229.86845,8.18351e+35,4.02629e+25,132985.25,1.6463145e+33,1.6009162e-29,2.2420042e+06,-1.8082302e+37,1.8547212e-25,-1.06024166e-19,-2.535604e+18,-4.3646336e+08,-9.950305e+29,-1.8882463e-07,-3.4157924e+20,-5.901858e+10,-0.18861994,-2.4086842e-14,0.003140079,-2.459434e+33,-4.860831e+15,-6.0094746e+13,4.6761316e+17,1.4877768e-17,1.7619126e+13,-4.488968e-25,-1.8169608e-37,0.00060694595,-6.389462e+37,-1.0688223e-27,5.7220565e-11,-5.2419953,1.3611266e+12,1.4334559e-17,-3.5772002e+06,2552.0608,-4.2158537e-14,-3.3542796e+20,2.7039208e-23,-1.1677768e-28,-0.023811718,-3.2946247e+16,-252.31873,3.456231e-12,4.2732014e+09,0.35307908,7.6711166e+37,2.3851477e-36,2.282067e-20,23759.176,6.808897e-18,2.2083532e-15,3.8116967e-05,-2.6551125e-25,7.45262e-06,-4.650334e-15,-6.551262e+32,-5.762721e-17,-2.6743323e+38,-5.9969397e+25,1.2732799e+07,-5.7653655e-20,-1.2332854e-38,6.9525948,-901.3351,3.753918e+17,4.0231176e-26,1.0927693e+38,-4.0948666e-26,-1.2411173e+25,-8.967325e+33,3.3512117e-29,-1.3521433e-05,2.8826556e+32,5.2139016e+11,-1.4549023e+17,-1.24304345e+33,-2.2248168e-33,-1.0188807e-29,-5.505188e-26,1.5684352e-17,7.470972e+33,-1.6626225e+06,9.571467e-31,-1.072974e+06,1.4149142e+06,2.5829603e-08,3.138604e+08,5.0535256e+13,-4.6057328e+20,1.4854664e-16,2.9216875e-33,-7.6241235e+24,6.3216577,-0.17625876,-1.9651656e+28,7.5954535e-25,3.5911816e+36,6.095752e+33,3.366743e-29,-6.006252e+12,-9.320409e+11,1.0495249e+14,-1.1130261e-21,-1.1196364e-06,-1.6946154e+10,-0.00046403465,1.6909723e-25,2.0223836e-06,1.4480721e+29,1.0961605e+24,3.0654348e-30,-0.00023432654,4.7594333e+20,3.7283506e-08,1.0511542e+20,23.200907,-3.4210046e+21,3.5898857e+09,2.8490262e+31,-0.012849343,-0.00021769323,-7.012904e-11,8.452471e+36,1.0891884e-33,-1.9836695e+33,2.7544075e-08,3.7667683e-10,2.7118668e-38,9.932161e+25,4.4953066e-10,-6.8797176e+11,6.9430846e-33,8.014342e-08,8.107997e-15,5.444265e-27,5.864707e+37,4.3112e-41,1.1328653e+23,-4.108576e+24,4.8275437e-05,2.0748679e+30,3.966957e-07,2.9704346e-32,1.4085517e-35,9.663481e-06,1.2851544e-08,-0.00041336502,6.5147767e+27,-2.3344376e-20,2.8328786e+22,-1.4942326e+35,1.0238783e-24,-757224.3,-3.599152e-32,3.0066427e+25,4.64347e-16,3.5753748e+06,-4.281629e-21,-5.744184e-32,-1.0499556e+13,3.6490373e-28,2.4247966e-21,7.5978903e-07,0.19106278,-8.769844e-35,-2.5574389e-05,1.2478965e+16,-8.106048e-29,4.566092e-09,6.848159e+19,7.3119107e+15,6.681818e+19,-1.78627e+31,0.00010021051,-6.965604,1.7332632e-31,0.00014148012,61031.883,-4.990052e-06,-686001.7,7.3856795e-29,3.3201342e-09,2.5299789e+08,-0.24247022,-5.2543266e+21,-7.774818e+25,3.8916078e+09,6.962151e+10,2.7672762e+37,-0.036339775,-1.3777967e+19,1.1393404e-27,9.600495e-11,8.1569513e+11,7404.232,3.438334e+09,2.5278484e-19,-4.631091e+15,1.4138528e+13,-1.2542953e+19,6.353923e-09,-5.577507e+24,-1.9478597e-10,5.5087836e-23,1.0978556e-22,5.6947947e-22,1.020288e+34,-5.207676e+33,-3.4508083e+21,1.6856275e-30,-1.5495471e-27,2.0690012e+19,2.8728028e-33,-13458.988,-5.280436e-24,0.0075049,5.0335716e-23,-5.9502625e+36,-1.5730207e-18,1.2734333e-21,3.7402296e-35,-8.268127e+19,3.2904477e-08,1.8042545e-36,-4.754868e-13,8.007872e+09,-1.6148839e-34,6.182925e+35,3.4752692e+10,3.1936394e+20,-8.650893e-29,8.3983627e+21,2.104372e-15,-1.04801615e-29,3.5188879e-22,1.4354661e-30,3.2781698e+36,-7.891975e-20,0.0007678309,5.23636e-20,-5.4031913e-25,2.2375539e-26,0.0016984894,8.250553e-38,1.7833659e+22,-4.386061e-35,-1.8419575e+06,0.57733643,-1.5424097e+33,1.6372937e-34,-7.4382754e-16,-2.5686286e+33,-4.5553403e-10,5.077788e+19,-8.005222e+37,5.336325e-12,-0.0038063033,-2.1582345e+18,-1.6632749e+33,-181.75269,8.803947e-30,9.7641886e-20,-1.7432177e+16,7.1051664,1.4761976e+31,-4.6200883e+30,1.6608307e-24,3.403859e-20,-2.4815951e-14,1.441379e+19,9.957967e-25,0.14609683,7.1300625e-23,-0.007148647,6.169929e+16,-2.449368e+31,1.5789136e-35,151.28781,-7.193823e-32,-4.7135978e+30,1.6778176e-15,4.1348974e-07,1.1781495e-38,1.3428846e-15,-3.9650965e+24,-1.4127561e-12,-3.1681673e-15,-1.0521047e-14,-195148.44,3.2529273e-32,-6.483802e+34,-4.1557117e+08,-4.7362623e+35,16.057388,-3.8137777e-32,3.691582e+12,5.213517e-15,7.317317e+28,-2.7551011e+35,-0.6669087,4.4463507e-33,0.049345456,220.378,138.20233,0.0900629,3321.1257,3.840011e-15,-5.173004e-21,4.1959355e-10,4.8876227e-06,4.38779e+25,4.691874e+14,3.4999095e-05,2.6063885e-27,-0.14058177,-1.31637064e+08,6.5142796e+32,-2.0592758e-20,2.4057582e-22,47.440037,2.2140072e+10,-0.7518098,-2.1641129e-29,-7.3128695e-17,3.050598e+06,-6.2334414e+29,-3.434132e-09,6.416509e+35,0.53068775,1.3511109e+38,-1.5864204e-23,-5.7759903e+24,-2.5711615e-25,2.5435424e-37,-5.5294756e-31,9.9816265e-26,8.951102e-17,9.111336e+17,-2.5349452e+20,-2.383783e-22,7.9118745e-22,-2.1398483e-34,-6.2789816e-34,7.7638226e-29,1.3500727e-30,-8.1179454e+12,-4.4044042e-23,0.0017816612,-3.1165378e+06,6.79456e-22,-1.01718e-39,-4.2429955e+21,-2.344386e+16,-3.0568253e+34,-6.370803e-19,2.2335302e-12,123.808395,2.3448742e-17,-0.00012088617,-4.4306766e+30,2.4486003e+08,-3.495928e-32,1.8249083e-34,-3.0039126e+35,-1.9275251e-16,-5.16431e-22,-2.2510224e-10,0.012862167,1.7583902e+35,-4.0444393e-26,-2.233208e+38,8.800437e+09,5.614796e+14,-6.4570635e-21,-2.9000888e+09,-6.401365e-10,6.7178894e+20,9.751177e-08,4.667077e+30,1.9062482e+16,3.2123182e+32,-1.986079e-31,7.249116e+33,-223.94817,-1.5636046e-20,-3.1796322e-36,-13.405595,-4.8670526e+25,-4.080441e-31,-2.2287253e-38,-1.6296579e+14,-1.9783353e+29,-640854.44,9.60665e-29,-1.0838821e-14,1.3184408e-21,9.0634903e+21,1.7517986e+22,1.7533954e-05,-1.1793142e-35,7.234647e-06,0.08926903,-8.556361e+29,-1.9294207e-22,4.0217345e+06,-8.745848e-25,2.768648e-23,-6.7127157e-25,8.339709e-33,-7.4667914e-19,1.6637033,1.472585e-27,-195962.58,-7.719408e-15,127151.734,85816.305,-6.155239e+20,0.083950564,-4.4922375e+14,-1.4836385e+34,-1.6362353e-25,1.126749e+31,3453.3843,-13.016247,4.0524406e-25,6.36164e+17,54483.688,-2.2694337e-10,-3.670832e-14,-2.6566173e-07,9.2820424e+30,2.1600447e+27,-5.25295e+14,3.491877e+32,7.12716e+33,-7.2944187e-26,-4.29328e+28,2.9178152e+07,5.439878e+32,-4.851176e-29,-5.4829486e-31,4.387904,5.9034872e-30,5.4459314e+37,3.4776046e-19,-4349.134,4.7029835e+14,359691.78,1.4191738e-33,3.5153393e+27,-1.2653031e+07,0.3673967,245.43224,1.0746286e+25,-1.7517876e-05,27362.555,-4.943882e-15,-2.8697792e-33,-7.5895797e-23,-3.6057367e+10,3.4581475e+08]':::VECTOR,'01060000C00600000001030000C0010000000A0000000F3F65DDA540E5C1BDFEFFEE93F6A2C1181951199659FDC1A8414BC41305EAC1DAA70AE4739BDFC131D9E1ED83CDEEC1096E5FEAC77FE3C1F43091DD1326F5C1AB3F4CC77768F1C1A0F5412B6FBE00C2C0C0E1B820D2F641D6802336583FEEC1CADDC9578168FB4153B53D03477F00C2E4A9FFA38D64C841864F1F38C9FDDCC146A88BF6F9C8F44152E7FE05EB15D841322DF88C1AE1F841542CDC0E1ED9E3C1E44AF299A8A2FC41624D0A95841CE14118AC286E4853A741305B20F90AD2F541E66344B0B3F6D041D69947BDA37EF941A55D646A17F80042F34931EDA798F2415AD6EFC96856F3C1B49E226FFA4402421700BA9B00FE9741214A22F5C553F3412A0250E361C5F3C1808346F1B913D6414FA5EF7F355CE2C13F74C15ACCDAE3410F3F65DDA540E5C1BDFEFFEE93F6A2C1181951199659FDC1A8414BC41305EAC101030000C0010000000A0000002E74B8145BFD01C2F609F212F4E600C231EA6E063723F4C1A6BB5B10717AFAC1BB94ED03D495F1C1584B4DA769B6F5C1CEA9D321FD27F7419BF4396FAF9FF04100E14D106B82E741DD6D30CF3551E9C1A58CD69EE5D7E6C120856FB61B7AEF41EB4D779F3ED3F041A2D004C00955FDC1D9A8DC6465A1F3C1DFB6A2006696F8C1AE459ED0DFF3F7415936B6BB102CEBC118B4F5EEF95EFDC17A3510703AC7FE41AE0FD2B0934DFC416E091E357BA9014291379044D8D8D9C126838DAF5DE4DC41BC502F33EFD6F741421A7366026FFE417F96AB69BEBAF4C1B78BB5263405EFC11C3F90FAB944F441B3C0CC92CF6CF7413FF710CEE911F1C10F99CDF6650AFAC169A79A856BF7F14167826570EAA4FF41707DFABEF1BDB441757357181753FAC12E74B8145BFD01C2F609F212F4E600C231EA6E063723F4C1A6BB5B10717AFAC101030000C00100000004000000947E5566987F01C2E517EFCC189FE441DDFEEECCFB33D5414E75C0DA0128E2418B909037ECEEE341BDE030658D65E341D955C3405919E8C124EA188D292CFF412330F79AF045FF4120432B1B88A3F3417AF717645937E141B6ADF7B80A76F441947E5566987F01C2E517EFCC189FE441DDFEEECCFB33D5414E75C0DA0128E24101030000C0010000000600000019859561D1F3ECC1F5843ABEEF1EFFC15960275FA90EDE418A0065E5FD66C94157C569F3FD7FB441BF66794ABFA6F1C1C3EFB0E843D8F4C18C813B4FFE59F0414BAFE519236BF64190A4219A4A1FF0C1890CB7A1830800429DB78D13764DC34159519CB49489F441886E3B17BA95FF414C8D2A78C2D1DE41BC286CF91796EC41B4035422A8BCEC419E171A1F438B0242AF9B10FE80E8EBC1E7B2AF6C8AADF04119859561D1F3ECC1F5843ABEEF1EFFC15960275FA90EDE418A0065E5FD66C94101030000C0010000000A0000004659F4201CFAD441708C5EB39574F4C17D2909C6D823E741395ED2AFB959CFC18F1328DF0364EA416A86A685420FE9C1D1493177FB32FA41A21DC2BAA490A4C1D31B57D3E586F9411375B1EE6B97D8C1841CFAF1510401C217BCAEE8027EF7C180DCA5345759014212D5E9EC77CEE3C1625415ECDEF1DF413DD3D162F334D6C1C5B02BF82C9ADD414B0A5C0AAC03EA41D45C35947145F8C1D2A21663C3F0F641A61F347E2535E3419DCD76EE006F0042C8BBE833E99BD341C8CF9360B5D0B441337869BA6BCFD3C127A2BF34778DF7415A004A4DFCF8ECC168D79233C111F0C1BA74F054A950FCC1B3C6B480802CEE416636DC3A1F30E8C1B635EBADE9C7F5419A8911ABAE9FF8C11E7B2136AD00DD41770B8BAD8A75F6C13650CDC845BFF9C14659F4201CFAD441708C5EB39574F4C17D2909C6D823E741395ED2AFB959CFC101030000C001000000040000009842D6BB5B800242A7BAE7DC8074C541A28A82D7E90AB3C1A8A0A2543DD700C29AD4A92DF860EE41E3438328A804F7410D24AE4E91570242C6474008B724ECC1241184B4EAC8C34187EA0AB90C20F4412EE7A69AB445F14185662BEE494BFA419842D6BB5B800242A7BAE7DC8074C541A28A82D7E90AB3C1A8A0A2543DD700C2':::GEOMETRY),(1083:::OID,e'w*\\x0f\\x0f\\b/'::NAME:::NAME,true,(-1.2781472206115723):::FLOAT8,'1983-01-25 22:28:36.000539':::TIMESTAMP,'BOX(-1.477061525460143 -2.114791394512738,0.6276522518799144 -0.4417721398608874)':::BOX2D,(-39):::INT8,1.4544307784306245:::FLOAT8,'[1.1445238e+08,1.8717871e-36,-4.9682172e+07,1.5201971e-15,-1.0416103e+25,-1.0457718e-12,7.9463124e-38,3.6166945e-31,2.4050353e+27,1.6556291e+31,1.4910051e-38,2.6505257e-13,-2.7448799e+09,-1.3413841e-14,-5.0299128e+26,8.2684884e-20,3.081292e-28,1.339452e+13,-9.695267e-13,-1.6298919e-34,1.6385612e-32,6545.0522,-1.2611738e+27,7.351525e+12,7.9244575e+25,-6.389244e+06,-5.705557e-21,1.2913123e-35,-6.5517624e-21,-4.512109,1.364498e-09,-7.772402e+25,1.1469373e+37,-4.4815766e-12,464359.3,-1.0007842e+09,1.2852066e-08,-6.982194e-17,7.2503714e-22,3.6716877e-10,-1.3830331e-15,-185.65172,-8.962583e-10,1.09319614e+33,-2.4124995e-31,-1.3721253e+16,2.9742592e-15,-2.3688783e-32,-5.106514e-14,5.420088e+06,1.6051892e+10,6.2470255e-23,-1.3964365e-08,3.2864039e-18,2.0443833e+23,-322.51163,-8.4071716e-10,-4.593323e-39,-3.4415645e+08,1.9598102e-38,1.5480791e+09,-1.8184805e+33,-9.849172e+30,-5.060523e-37,4.8216545e-23,8.1422723e-07,4.2144416e-25,-9.5135e+32,2.48414e-22,-1.9047574e-37,1.3490496e-09,-3.5696204e-10,-1.3157801e+08,52628.95,-1.1844344e-27,-1.501805e+25,9.2715346e-27,-19.51769,-1.26932e-23,-4.533175e+37,3.9526653e+35,3.827551e+27,-1.4624576e-09,1.4588001e-05,3.4409453e-26,4.2800075e+06,3.424611e+08,-3136.0159,-2.7385058e-17,7.658351e+22,-5.8953638e+29,-2.9540048e+29,1.4702496e-19,2.865471e+29,4.830411e+12,1.0952149e+30,-5.004824e+10,-1.0495219e+24,8.5397554e+15,-7.392933e+24,4.3611178e+24,6.427462e-16,1.787939e-19,-3606.1973,-12.888924,-3247.709,7.7202333e-19,-1.0713426e+35,1.6504366e+14,-2.1077427e+33,1.5733805e-07,0.002376407,-2.787581e+28,-4.5491505e+06,8.142165e+07,-6.1183945e-33,-3.8217337e+25,2.2517397e-15,9.7435446e-20,2.6297442e-20,1.5417405e-32,8.2924634e+18,-3.6393176e-06,570905.8,-9.518833e-13,2.9401447e-38,-6.222866e-17,1.4548508e+20,0.00013519562,-509504.72,3.0238242e-11,-3.254449e-11,-3.9136185e-26,2.7798885e-06,-7.512032e+31,-4.9863887e+25,1.0831721,-9.745128e+19,-1.6344652e+27,-0.051907383,7.136728e-21,-1.0010778e+21,-275.49347,6.716326e-07,2.0784185e-38,3.914251e+19,-3.4622731e-09,-2.2625549e-36,-3.1953024e+38,-3.2228665e-05,-110.511116,-7.487002e-14,-2.7164872,-3.2759857e-24,-1.6335903e-28,-3.665377e+13,1.4929356e+30,0.014953572,1.4088315e-18,1.968868e-12,1.646456e-37,1.4379203e+20,-1.34408735e+20,2.3514202e-21,2.7663913e+32,-4.995517e-17,-1.239536e-06,-5.4265994e-24,1.2199044e-38,-2.7309803e+21,6.734965e+30,-3.655742e+14,1.2893953e+22,1.1183847e+18,3.122638e-14,-8.230599e-33,-5.2301454e+14,3.888669e+32,1.2933076e+17,3.2943478e+34,-1.1497656e-20,1.4707886e+35,1.5312268,-1.8921037e-17,-3.359744e+17,1.1899098e-10,2.2060879e-21,-8.436422e+19,3.7149877e-28,-2.2016023e-14,-9.133067e+08,3.4112123e-11,4.0151654e-23,2.9287096e+09,1.2841559e+36,6.102322e-09,1.7208415e-36,2.2877168e+33,-9.486079e-35,-1.755757e+29,-3.108037e+33,-8.06455e-15,-1926.7014,-1.7199774e-27,-1.1244695e+32,0.00034515397,3.1752785e+11,7.923053e+10,-5.171523e-19,-4.8003973e-10,-9.505444e-06,-4.154703e-32,6.9849456e+15,2.5957669e+17,5.2347053e+23,1.0085192e-35,-1.1613107e+24,4.728138e-07,1.1135343e-17,1.2906261e-31,1.3743959e-09,-7.05299e-31,-16610.047,-1.377222e+12,-9.0693634e-38,-1.6830384e+06,1.0949531,-3.3272678e+16,-1.9989944e+27,-1.4668857e+21,-4.0014896e-13,1.4644156e-15,2.046505e-32,4.207046e-15,1.4605682e+31,-7.8269895e+37,0.00823284,-1.8006344e+37,-0.63049954,-2.8042962e-14,-8.547326,-1.3351378e-15,6.937812e+09,1.1032975e+07,1.4677202e+27,-1.7523051e-09,-4.2965087e-16,-7.408468e-23,9.268138e+37,-5.087045e+24,-1.4689039e-29,2.9566518e+35,-8.065057e+13,2.4229629e-17,8.3869614e+15,2.8368622e+35,1.1432178e+36,6.782022e+22,-1.5438336e+25,180.57498,-4.1999237e+37,1.8846822e+22,1.1593218e+13,4.051093e+31,-1.02863514e+37,39637.902,8.762849e-13,-1.3126634e+26,0.20130202,-1.337131e+23,-1.4620563e-13,-7.411326e+22,1.9419627e+24,202.20523,3.7651526e+29,1.0609359e-07,9.917907e-28,8.1807634e-23,28.617268,1.0227686e-30,0.004268859,-1.340095e+32,-2.9130346e+26,1.4028689e-14,-3.054218e+09,-3.0350637e+35,1.2311098e-32,-1.2927443e+33,-3.6420108e+14,-3.41224e+06,5.93147e-11,-1.3428497e+38,-8.1459067e+18,-2.59496e-18,-4.6806258e+32,-1.9494447e+11,-2.2334124e+35,181.77837,-1.1207775e-18,2.5063182e+08,-9.4342475e+36,-3.463836e-13,3.6982752e+36,-6.680127e+36,-3.8297633e-19,1.0955117e+27,5.0566673e-38,7.179736e+12,7.8690314e+37,5.7280135e+21,9.2511485e-24,5.1112827e-21,4.228715e-16,3.1811643e-37,2.4366197e-36,-4.312957e-21,-7487.88,1.24859415e-29,1.13920995e+26,3.6963303e-07,-1.0971206e+18,-2.2785602e-22,-3.1807334e-33,-1.434553e+29,-3.4563775e+26,3.25986e+33,5.5250485e+26,-1.8280006e+08,-1.2372841e+22,8.470354e-09,5.542991e-05,-3.0813436e+12,1.8880031e-11,1.2765743e-09,6.8768464e+31,-1.5677032e-06,-6.264877e-11,3.8162e-29,-2.6764528e-08,-2.6938376e-31,-1.6083876e+11,-9.482372e+28,-3.2902298e+19,1.3898896e-29,0.0015162781,2.7809508e-09,1.256795e+17,-1.7522008e-17,-1.775194e+14,-2.923307e+13,8.2050146e+30,8.489177e-37,2.3333587e-17,8.498028e+36,-6.8897408e+13,2.9220727e+20,-3.1362461e-13,2.3546866e-31,1.0570321e+12,4.1182758e-30,-2.1921518e-16,4.0017314e+21,-4.3424275e-31,219491.75,-2.4627402e+06,-1.0644212e+25,5.6442976e+23,-6.403783e+13,-3.631229e-16,5.55761e+18,-3.5836505e+34,-5.4759635e-08,1.2883639e+20,-3.5882087e+12,0.94572395,-1.7976084e+23,-5.566997e+23,6.7359425e+10,-8.3744916e-20,-7.5446663e-19,4.1529773e-35,1.36566505e-11,1.5809425e-30,-1.2630626e-06,18.137346,1.532834e-09,-1.781548e+09,-4.71789e-14,-0.00033545893,-9.369043e+16,-3.952307e-12,65.61634,1.085279e-31,8.218249e+16,1.9645428e-28,-6.590225e-16,7.943889e+09,1.9994116e-22,2.626274e+23,-9.760757e-10,4.433875e+34,3.885002e+17,-7.70174e+06,-1.9096506e+06,143045.14,4.8485882e-21,-6.2797534e+28,3.306336e+33,5.1466427e-17,8.416416e+20,-6.218242e+28,-3.836153e-28,-3.4773572e-38,0.23519804,1.3308382e+18,2.1382215e-08,21618.738,3.3414725e+16,1.891855e-06,-2.3875598e+18,-1.0174775e+37,1.1739762e-16,-5.3958533e+21,-3.4737247e+18,-336800.12,4.751096e-28,3.3276818e-36,-1.5491556e-21,-472.17548,-0.0011970467,-8.7773364e+15,2.0589547e+13,-3.5008096e-19,-1.1750497e-33,-5.2949854e+13,1.0541803e+33,-9.744696e-21,-8.165728e+10,1.2186533e+07,-3.5157552e-19,1.6468696e+19,1.7400304e+17,-4.5116832e-33,6.459476e-25,783.0909,-23528.12,7.1249973e-37,-0.00016519269,-1.6027835e-28,224710.94,2.7224778e+14,4.566188e+16,0.0001558111,7.7249624e-31,2.9236153e+32,3.8165283e-38,1.9208031e+28,-1.7377862e+36,-1.7891358e+36,4.5598183e+10,5.939123e-18,-8.103058e-35,-1.4353651e+10,1.2266488e+12,-1.5702497e-34,1.5521693e+27,3.3716323e-14,0.016347012,2.7385357e+14,8.602483e-07,4.2263747e-21,-2.4325467e-15,3.1391337e+35,7.414264e-07,-1.0676118e-25,-4.0646023e-29,0.08568816,0.033148997,2.1708672e+22,3.181811e+08,1.9709939e+37,-1.9207808e-27,-1.494912e+37,-2.4318223e+28,2.6879142e-35,5.616917e+21,-0.00021531463,-1.4824042e+36,2.088085e-29,-1.1156685e-15,-5.010244e+20,0.38188028,-7702.5083,1.8883066e-17,0.0030047877,1.86171e-24,5.4277607e+12,1.2047788e+32,7.451955e+33,6.3297545e+23,-1.4465839e+28,1.2954274e+19,-131.48502,-1.0084146e+32,-0.0064417818,-1.4849633e+27,-836.9132,5.2709464e+29,-4.76543e-11,0.0019756746,-4.984318e+35,1.0863658e+09,-53.93503,-2.2129182e+22,1.03908435e-23,-1.257451e+18,1.2824674e+32,-0.00022212474,1.3157537e-08,0.48568043,1.6582752e+16,1.6482999e+11,15.186034,1.0871877e-10,-2.1953927e-33,1.8043887e-22,9.7915546e+26,-9.626768e+32,7.607765e+27,-1.0259547e-08,-1.9617592e+09,-2.155289e-08,-6.4043635e-21,7.2643415e-19,1.2579014e+11,5.4159697e-19,-4.293576e-14,-6.724874e+20,-6.5888177e-22,-3.8801657e-38,4.694166e-25,8.8280656e-29,1.6839135e-38,8.058726e+32,-2.0761342e-24,-3.9961825e-23,-8.819766e+29,2.4174907e-23,-1.0220364e+07,-3.4443814e-33,107145.84,1.7512158e+37,-1.1006235e-09,-7.8798126e+21,-6.331724e-14,-7.027874e-11,-5.302507e+15,-8.787591e+31,2.8312548e-15,1.845407e+21,-1.7350936e+25,0.0007281423,4.331293e-32,-9.0240746e+27,3.8243665e-27,4.9287803e-34,-6.9742174e-07,0.01089251,2.2792364e+20,-49371.816,7.0900626e+37,-5.7178125e+08,-1.6489009e+33,-4.4595654e-26,-1.657219e+18,-1.2253694e-06,-5.6840958e+22,-9.0287634e-27,3.5525278e-09,4.6052032e-10,-4.437617e+13,6.4236836e-33,6.124909e+36,-0.0007656525,-4.041291e+13,-2.6807869e-20,-4.875483e-34,-1.0425698e-21,2.999114e-13,2.601442e+21,-18307.64,7.183522e+19,-8.081033,3.2502793e+19,-3.3750268e+06,4.3413014e-15,-1.7575407e-26,3.7667986e-13,1.9620099e-31,-1.3123354e+31,-9.159005e-36,-9.261928e+14,-1.4771628e+35,3.0464065e+14,1.021646e+18,2.6470255e+20,-6.1843443e-38,-0.0037308196,1.5467159e+37,6.344762e-16,-6.446354e-31,-7.103233e+36,-0.36875698,-6.4382583e-34,-1.14741394e+36,1.9418365e+16,130.40558,-2.3598488e-09,2.8531459e+09,4.5054507e+18,-7.345739e+19,-7.422284e-05,-1.8177688e+29,2.8098746e+34,-9.807493e+23,-6.4249677e-19,2.2377308e+20,278742.7,5.116355e-28,-1.1766771e+30,-3.0776896e+09,-5.06526e+13,4.5616813e+08,7.269297e-33,-1.5467168e+32,1.5717372e-27,-1.4989407e-05,-1.05383905e-23,7.163641e+35,-43.11378,-2.8116463e-30,-6.9531064e-20,1.6717022e+33,1.6870352e-28,5.5199715e+06,-3.7404148e+07,-3.4258454,2.3923373e-30,-2.7126508e+17,703340.75,3.6871888e+06]':::VECTOR,'01020000C000000000':::GEOMETRY)",
   "CREATE TABLE schema_w0_13.table_w0_14 (\"c ol14_w0_15\" INTERVAL NOT NULL, col͍14_w0_16 NAME NULL, PRIMARY KEY (\"c ol14_w0_15\" DESC), UNIQUE (\"c ol14_w0_15\" ASC, col͍14_w0_16 DESC), UNIQUE (col͍14_w0_16 ASC, \"c ol14_w0_15\"), UNIQUE (col͍14_w0_16), INVERTED INDEX (col͍14_w0_16 gin_trgm_ops ASC), INVERTED INDEX (\"c ol14_w0_15\", col͍14_w0_16 gin_trgm_ops ASC), INDEX (lower(col͍14_w0_16) ASC) NOT VISIBLE, UNIQUE ((CASE WHEN \"c ol14_w0_15\" IS NULL THEN '':::STRING ELSE 'xCk|_GG':::STRING END) DESC, col͍14_w0_16 DESC), FAMILY (col͍14_w0_16, \"c ol14_w0_15\"))",
   "DROP SCHEMA \"schema_w1_1\" CASCADE"
  ]
 }
}
    schema_change_external_test.go:133: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:133
        	Error:      	Received unexpected error:
        	            	***FAIL; Failed to receive an execution error when errors were expected
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3016
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:499
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:587
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:122
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1222
        	            	Wraps: (4) ***FAIL; Failed to receive an execution error when errors were expected
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 36
Total Schema Statements Succeeded = 36
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 16
Total Transactions Rolled Back = 26
Total Transactions Executed = 42
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload3319425535
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload3319425535
--- FAIL: TestWorkload (6.64s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-40950

--- a/pkg/workload/schemachange/error_screening.go
+++ b/pkg/workload/schemachange/error_screening.go
@@ -1285,31 +1285,18 @@ func (og *operationGenerator) schemaContainsHasReferredFunctions(
 	return len(result) > 0, err
 }
 
-func (og *operationGenerator) schemaContainsTypesWithCrossSchemaReferences(
+func (og *operationGenerator) schemaContainsTypes(
 	ctx context.Context, tx pgx.Tx, schemaName string,
 ) (bool, error) {
 	ctes := []CTE{
 		{"descriptors", descJSONQuery},
-		{"functions", functionDescsQuery},
 		{"types", enumDescsQuery},
-		{"referenced_descriptor_ids", `
-				SELECT json_array_elements(descriptor->'referencingDescriptorIds')::INT8 AS id FROM types WHERE schema_id = $1::REGNAMESPACE::INT8
-			UNION ALL
-				SELECT (ref->'id')::INT8 AS id FROM (SELECT json_array_elements(descriptor->'dependedOnBy') AS ref from functions WHERE schema_id = $1::REGNAMESPACE::INT8)
-		`},
 	}
 
-	_, err := Collect(ctx, og, tx, pgx.RowToMap, With(ctes, `SELECT * FROM types WHERE schema_id = $1::REGNAMESPACE::INT8`), schemaName)
+	result, err := Collect(ctx, og, tx, pgx.RowToMap, With(ctes, `SELECT * FROM types WHERE schema_id = $1::REGNAMESPACE::INT8`), schemaName)
 	if err != nil {
 		return false, err
 	}
-
-	result, err := Collect(ctx, og, tx, pgx.RowToMap, With(ctes, `
-		SELECT $1::REGNAMESPACE::INT8 AS this_schema_id, * FROM descriptors d
-		WHERE schema_id != $1::REGNAMESPACE::INT8
-		AND EXISTS(SELECT * FROM referenced_descriptor_ids WHERE id = d.id)
-		AND (NOT descriptor ? 'table')
-	`), schemaName)
 	return len(result) > 0, err
 }
 

--- a/pkg/workload/schemachange/operation_generator.go
+++ b/pkg/workload/schemachange/operation_generator.go
@@ -3910,10 +3910,10 @@ func (og *operationGenerator) dropSchema(ctx context.Context, tx pgx.Tx) (*opStm
 	if err != nil {
 		return nil, err
 	}
-	crossReferences := false
 	crossSchemaFunctionReferences := false
+	schemaHasTypes := false
 	if schemaExists {
-		crossReferences, err = og.schemaContainsTypesWithCrossSchemaReferences(ctx, tx, schemaName)
+		schemaHasTypes, err = og.schemaContainsTypes(ctx, tx, schemaName)
 		if err != nil {
 			return nil, err
 		}
@@ -3927,8 +3927,13 @@ func (og *operationGenerator) dropSchema(ctx context.Context, tx pgx.Tx) (*opStm
 	stmt.expectedExecErrors.addAll(codesWithConditions{
 		{pgcode.UndefinedSchema, !schemaExists},
 		{pgcode.InvalidSchemaName, schemaName == catconstants.PublicSchemaName},
-		{pgcode.FeatureNotSupported, crossReferences && !og.useDeclarativeSchemaChanger},
 		{pgcode.DependentObjectsStillExist, crossSchemaFunctionReferences && !og.useDeclarativeSchemaChanger},
+	})
+	stmt.potentialExecErrors.addAll(codesWithConditions{
+		// TODO(spilchen): we need this if dropping a schema that has a type
+		// referenced in another schema. This can be removed once issue #51480 is
+		// addressed.
+		{pgcode.FeatureNotSupported, schemaHasTypes},
 	})
 
 	stmt.sql = fmt.Sprintf(`DROP SCHEMA "%s" CASCADE`, schemaName)

--- a/pkg/workload/schemachange/optype.go
+++ b/pkg/workload/schemachange/optype.go
@@ -297,7 +297,7 @@ var opWeights = []int{
 	createView:                        1,
 	dropFunction:                      1,
 	dropIndex:                         1,
-	dropSchema:                        0, // Disabled and tracked with #127977.
+	dropSchema:                        1,
 	dropSequence:                      1,
 	dropTable:                         1,
 	dropView:                          1,

